app/models/Site.java

```diff
@@ -32,6 +32,6 @@ public static Site findById(String id){
         // Craig, I've started this for you, this returns all clients.
         // You need to change this to bring back only the associated clients
         // to this site, making using of the clientIds property of this class.
-      return (List<Client>) Base.toList(Client.clientsCollection.find().sort(sortString).as(Client.class));
+      return (List<Client>) Base.toList(Client.clientsCollection.find("{site:#"}, this.siteId).sort(sortString).as(Client.class));
```

Comment:
This isnt correct either.....I've done a quick bit of research. I'll work on it this week and see what is wrong.